### PR TITLE
Sprint12 04 a44375 comments update

### PR DESCRIPTION
**Pull request by Neve**
Enormous quantity of new code comments
some dead code removed

### CODE REVIEW

File: cookbooks/app/recipes/do_loadbalancers_allow.rb [hunk 1]
```diff
@@ -11,6 +11,7 @@ class Chef::Recipe
   include RightScale::App::Helper
 end
 
+# Adding iptables rule to allow loadbalncers <-> application servers connections
```
Comment (douglaswth) [inline cookbooks/app/recipes/do_loadbalancers_allow.rb]: It should be spelled "loadbalancers".

File: cookbooks/app/recipes/do_loadbalancers_deny.rb [hunk 2]
```diff
@@ -11,6 +11,7 @@ class Chef::Recipe
   include RightScale::App::Helper
 end
 
+# Adding iptables rule to disable loadbalncers <-> application servers connections
```
Comment (douglaswth) [inline cookbooks/app/recipes/do_loadbalancers_deny.rb]: It should be spelled "loadbalancers".

File: cookbooks/app_passenger/attributes/default.rb [hunk 3]
```diff
@@ -24,14 +28,21 @@
...
+# Path to Ruby EE gem directory
 set[:app_passenger][:ruby_gem_base_dir]="/opt/ruby-enterprise/lib/ruby/gems/1.8"
+# Path to Ruby EE gem executable
 set[:app_passenger][:gem_bin]="/opt/ruby-enterprise/bin/gem"
+# Path to Ruby EE ruby executable
 set[:app_passenger][:ruby_bin]="/opt/ruby-enterprise/bin/ruby"
+# Path to passenger module for apache
 set[:app_passenger][:apache_psr_install_module]="/opt/ruby-enterprise/bin/passenger-install-apache2-module"
-
+# By default rails application environment variable will is set to "development"
```
Comment (douglaswth) [inline cookbooks/app_passenger/attributes/default.rb]: I don't think the "will" should be there.

File: cookbooks/app_passenger/providers/default.rb [hunk 4]
```diff
@@ -66,7 +57,7 @@
     only_if do node[:kernel][:machine].include? "i686" end
     cookbook 'app_passenger'
   end
-
+  # Unpack sources. RubyEE ready to action.
```
Comment (douglaswth) [inline cookbooks/app_passenger/providers/default.rb]: What does "ready to action" mean? Also, please don't remove the whitespace between resource calls.

File: cookbooks/app_php/templates/default/db.php.erb [hunk 5]
```diff
@@ -1,4 +1,12 @@
 <?php
+//
+// Php database connection Config file, version 1.
```
Comment (douglaswth) [inline cookbooks/app_php/templates/default/db.php.erb]: PHP is usually all capitals. Do we need a version number for this?

File: cookbooks/app_tomcat/providers/default.rb [hunk 6]
```diff
@@ -51,19 +54,20 @@
...
   execute "alternatives" do
     command "#{node[:app_tomcat][:alternatives_cmd]}"
     action :run
   end
 
+  # Installing database adapter for tomcat
   db_adapter = node[:app_tomcat][:db_adapter]
   if db_adapter == "mysql"
-    # Link mysql-connector plugin to Tomcat6 lib
+    #Removing existing links to database connector
```
Comment (douglaswth) [inline cookbooks/app_tomcat/providers/default.rb]: Please add a space between the # and the begging of the sentence.

File: cookbooks/app_tomcat/providers/default.rb [hunk 7]
```diff
@@ -168,7 +174,7 @@
     cookbook 'app_tomcat'
   end
 
-
+    # Starting tomcat service
     action_start
```
Comment (douglaswth) [inline cookbooks/app_tomcat/providers/default.rb]: Why is this code indented further than its surrounding code?

File: cookbooks/repo_git/libraries/helper.rb [hunk 8]
```diff
@@ -5,33 +5,45 @@
 # RightScale Terms of Service available at http://www.rightscale.com/terms.php and,
 # if applicable, other agreements such as a RightScale Master Subscription Agreement.
 
+
+
```
Comment (douglaswth) [inline cookbooks/repo_git/libraries/helper.rb]: Was this extra space really necessary?

File: cookbooks/repo_git/libraries/helper.rb [hunk 9]
```diff
@@ -5,33 +5,45 @@
...
   module Repo
     class Ssh_key
      KEYFILE = "/tmp/gitkey"
 
-     # Add ssh key and exec script
+     # Create bash script, which will set user defined ssh key required to access to private git source code repositories.
+     # Parameters:
+     #   ssh_key(String):: Git private ssh key
+     # Raise:
+     #   RuntimeError:: if ssh key string is empty
```
Comment (douglaswth) [inline cookbooks/repo_git/libraries/helper.rb]: Please use YARD syntax for this kind of stuff. The parameter should be `git_ssh_key` according to the code.

File: cookbooks/repo_git/recipes/default.rb [hunk 10]
```diff
@@ -14,9 +14,12 @@
 else
   package "git"
 end
-
+# Install additional git packages
+# Generalized Interface Toolkit
```
Comment (douglaswth) [inline cookbooks/repo_git/recipes/default.rb]: I don't think that "Generalized Interface Toolkit" is the `gitk` that this is installing. `gitk` is a Git UI tool. I'm not sure why it is being installed on a server though.

Comment (Neve) [inline cookbooks/repo_git/recipes/default.rb]: http://stackoverflow.com/questions/1516720/git-gui-client-for-linux/1518844#1518844

http://gitk.sourceforge.net/

We do not use this recipe in ST's

File: cookbooks/app_php/templates/default/db.php.erb (hunk 5 above)
Comment (Neve) [inline cookbooks/app_php/templates/default/db.php.erb]: https://wookiee.rightscale.com/display/rightscale/ServerTemplate+Development+Guide#ServerTemplateDevelopmentGuide-Templates

File: cookbooks/repo_git/recipes/default.rb (hunk 10 above)
Comment (Neve) [inline cookbooks/repo_git/recipes/default.rb]: I dont think that it is really necessary
but this package was in recipe from early beginning
however I can remove it

File: cookbooks/app_php/templates/default/db.php.erb (hunk 5 above)
Comment (douglaswth) [inline cookbooks/app_php/templates/default/db.php.erb]: Your section does not seem to be actually following that template and the version number in the template is 4.x which relates to collectd there.

Comment (Neve) [inline cookbooks/app_php/templates/default/db.php.erb]: k will remove it
in next commit

Comment (Neve) [inline cookbooks/app_php/templates/default/db.php.erb]: update commited

File: cookbooks/repo_git/recipes/default.rb (hunk 10 above)
Comment (douglaswth) [inline cookbooks/repo_git/recipes/default.rb]: Can you please remove it? Thanks!

Comment (Neve) [inline cookbooks/repo_git/recipes/default.rb]: done

File: cookbooks/repo_git/libraries/helper.rb [hunk 11]
```diff
@@ -10,28 +10,41 @@ module Repo
...
      KEYFILE = "/tmp/gitkey"
 
-     # Add ssh key and exec script
+     # Create bash script, which will set user defined ssh key required to access to private git source code repositories.
+     #
+     # == Parameters
+     # git_ssh_key (string):: Git private ssh key
+     #
+     # == Raise
+     # RuntimeError:: if ssh key string is empty
```
Comment (douglaswth) [inline cookbooks/repo_git/libraries/helper.rb]: This is still not in YARD format.

**Comment by douglaswth** [comment]
This looks ready to merge.

**Pull request merged**